cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,227 @@
+                    alterQuotas(reconciliation, vertx, kafkaAdmin, clientQuotaAlteration)
+                        .onComplete(result -> {
+                            if (result.succeeded()) {
+                                LOGGER.infoCr(reconciliation, "Successfully altered default user quotas");
+                                promise.complete();
+                            } else {
+                                LOGGER.errorCr(reconciliation, "Failed to alter default user quotas", result.cause());
+                                promise.fail(result.cause());
+                            }
+
+                            kafkaAdmin.close();
+                        });
+
+                    return promise.future();
+                } else {
+                    kafkaAdmin.close();
+                    return Future.succeededFuture();
+                }
+            });
+    }
+
+    /**
+     * Checks whether default quotas configuration needs to be altered with new configuration.
+     * It gets the current default quotas for users from Kafka and compares them with the desired configuration
+     *
+     * @param reconciliation          Reconciliation marker
+     * @param vertx                   Vert.x instance
+     * @param kafkaAdmin              Kafka Admin object
+     * @param ops                     List of {@link ClientQuotaAlteration.Op}
+     * @param isNotKafkaPlugin        boolean parameter determining if the Kafka built-in quotas plugin is used
+     *
+     * @return  result determining if the new quotas configuration should be applied or not
+     */
+    /* test */ static Future<Boolean> shouldAlterDefaultQuotasConfig(Reconciliation reconciliation, Vertx vertx, Admin kafkaAdmin, List<ClientQuotaAlteration.Op> ops, boolean isNotKafkaPlugin) {
+        return VertxUtil.kafkaFutureToVertxFuture(reconciliation,
+                vertx,
+                kafkaAdmin.describeClientQuotas(ClientQuotaFilter.containsOnly(List.of(ClientQuotaFilterComponent.ofDefaultEntity(ClientQuotaEntity.USER)))).entities())
+            .compose(clientQuotaEntityMapMap -> {
+                Map<String, Double> currentQuotas = clientQuotaEntityMapMap.get(DEFAULT_USER_ENTITY);
+
+                if (currentQuotas == null) {
+                    // 1. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is not of type `QuotasPluginKafka`, skip alteration
+                    // 2. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is type of `QuotasPluginKafka`, but there is no fields configured, skip alteration
+                    if (isNotKafkaPlugin || ops.stream().allMatch(op -> Objects.isNull(op.value()))) {
+                        LOGGER.debugCr(reconciliation, "There are no default user quotas set in Kafka and the Kafka built-in plugin is not configured, skipping the alteration");
```

Comment:
Not sure we need this one-of log message. You should either log all the outcomes here, or maybe better remove this and add the outcomes to what suggested above.

Reply:
I was thinking about splitting the `isNotKafkaPlugin || ops.stream().allMatch(op -> Objects.isNull(op.value()))`
to `if` and `else if` where I will provide a correct debug message -> because adding all the outcomes to the one suggested little bit above would be quite long and during the debug we would not see the exact branch/path it took.